lib/gossip/state_transitions.js

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var Member = require('../membership/member');
+
+function StateTransitions(options) {
+    this.ringpop = options.ringpop;
+
+    var p = options.periods || {};
+    this.suspectTimeout = p.suspect || StateTransitions.Defaults.suspect;
+
+    this.logger = this.ringpop.loggerFactory.getLogger('statetransitions');
+
+    this.enabled = null;
+    this.timers = {};
+}
+
+StateTransitions.prototype.enable = function enable() {
+    if (this.enabled !== true) {
```

Comment:
`enabled` seems to be used throughout the code in a way that does not make sense for the name of the application. where `enabled` is set to `true` to **disable** the module.